src/test/java/de/tum/in/www1/artemis/util/CourseTestService.java

```diff
@@ -731,6 +731,51 @@ public void testGetCoursesWithoutActiveExercises() throws Exception {
         assertThat(courseInList.getExercises()).as("Course doesn't have any exercises").isEmpty();
     }
 
+    // Test
+    public void testGetCoursesRegisteredUnregisteredStudentExam() throws Exception {
+        User student = userRepo.findOneWithGroupsByLogin(userPrefix + "student1").get();
+
+        String suffix = "active";
+        adjustUserGroupsToCustomGroups(suffix);
+        Course course = ModelFactory.generateCourse(null, null, null, new HashSet<>(), userPrefix + "student" + suffix, userPrefix + "tutor" + suffix,
+                userPrefix + "editor" + suffix, userPrefix + "instructor" + suffix);
+        course = courseRepo.save(course);
+        Exam examRegistered = ModelFactory.generateExam(course);
+        examRegistered.addRegisteredUser(student);
+        examRepo.save(examRegistered);
+        Exam examUnregistered = ModelFactory.generateExam(course);
+        examRepo.save(examUnregistered);
+        List<Course> courses = request.getList("/api/courses/for-dashboard", HttpStatus.OK, Course.class);
+        final var finalCourse = course;
+        Course courseInList = courses.stream().filter(c -> c.getId().equals(finalCourse.getId())).findFirst().orElse(null);
+        assertThat(courseInList).isNotNull();
+        assertThat(courseInList.getExams()).hasSize(1);
+        assertThat(courseInList.getExams()).contains(examRegistered);
+    }
+
+    // Test
+    public void testGetCoursesInstructorExam() throws Exception {
+        User student = userRepo.findOneWithGroupsByLogin(userPrefix + "student1").get();
+
+        String suffix = "active";
+        adjustUserGroupsToCustomGroups(suffix);
+        Course course = ModelFactory.generateCourse(null, null, null, new HashSet<>(), userPrefix + "student" + suffix, userPrefix + "tutor" + suffix,
+                userPrefix + "editor" + suffix, userPrefix + "instructor" + suffix);
+        course = courseRepo.save(course);
+        Exam examRegistered = ModelFactory.generateExam(course);
+        examRegistered.addRegisteredUser(student);
+        examRepo.save(examRegistered);
+        Exam examUnregistered = ModelFactory.generateExam(course);
+        examRepo.save(examUnregistered);
+        List<Course> courses = request.getList("/api/courses/for-dashboard", HttpStatus.OK, Course.class);
+        final var finalCourse = course;
+        Course courseInList = courses.stream().filter(c -> c.getId().equals(finalCourse.getId())).findFirst().orElse(null);
+        assertThat(courseInList).isNotNull();
+        assertThat(courseInList.getExams()).hasSize(2);
+        assertThat(courseInList.getExams()).contains(examRegistered);
+        assertThat(courseInList.getExams()).contains(examUnregistered);
```

Comment:
```suggestion
        assertThat(courseInList.getExams()).containsExactly(examRegistered, examUnregistered);
```
Maybe it’s called `containsElementsInAnyOrder`? Exactly asserts the order, too, I think.